app/controllers/HasPAYEController.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.FrontendAppConfig
+import identifiers.TypedIdentifier
+import models.Mode
+import models.requests.DataRequest
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{AnyContent, Result}
+import services.UserAnswersService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.{Navigator, UserAnswers}
+import viewmodels.CommonFormWithHintViewModel
+import views.html.hasPAYE
+
+import scala.concurrent.Future
+
```

Comment:
Well, we should rename the HasCrnController if we're going to do that as it doesn't make logical sense to have paye controller extending crn controller trait. Also, the view is slightly different so this will need to be passed to the parent class somehow.

Reply:
Yes, we should rename it and pass all the view details by means of viewmodel

Reply:
Fixed